### PR TITLE
use versioneer

### PR DESCRIPTION
**Pull request by ocefpaf**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -3,30 +3,7 @@
...
-        TestCommand.finalize_options(self)
-        self.verbose = True
-
-    def run_tests(self):
-        import pytest
-        errno = pytest.main(self.test_args)
-        sys.exit(errno)
-
-
-def extract_version(module='utide'):
```
Comment (ocefpaf) [inline setup.py]: Most of the PR is bolierplate added automatically by `versioneer`. We need to do that only once.
I removed this function b/c it is obsolete now.

File: setup.py [hunk 2]
```diff
@@ -3,30 +3,7 @@
 from setuptools import setup
 from setuptools.command.test import test as TestCommand
 
-
-class PyTest(TestCommand):
```
Comment (ocefpaf) [inline setup.py]: We don't really use this class, removing for simplicity.

File: utide/__init__.py [hunk 3]
```diff
@@ -20,8 +17,12 @@
 constit_index_dict = dict([(name, i) for (i, name) in
                            enumerate(constit_names)])
 
-from ._solve import solve  # noqa
-from ._reconstruct import reconstruct  # noqa
+from ._solve import solve
+from ._reconstruct import reconstruct
```
Comment (wesleybowman) [inline utide/__init__.py]: These failed PEP8 before, but now they are good?

Comment (ocefpaf) [reply utide/__init__.py]: Yep. flake8 was updated to understand those are used below in the `__all__` variable.

**Comment by wesleybowman** [comment]
Merging

File: utide/__init__.py (hunk 3 above)
Comment (wesleybowman) [reply utide/__init__.py]: Another quick question then, can we remove the allowed failures in Travis now?

Comment (ocefpaf) [reply utide/__init__.py]: I'll work on that, there are some real pep8 infringements.
Note that, although I do like to follow code standards, I usually don't let those be real failures and drive the attention away from the code in PRs, so I recommend to leave them in the allowed failures.

Comment (wesleybowman) [reply utide/__init__.py]: Yeah, I can agree on that, but I was more wondering why 3.6 passes and only 3.5 doesn't

Comment (ocefpaf) [reply utide/__init__.py]: We only run `coding_standards` on Python 3.5, no need to run that more than once.

**Pull request merged**